src/modules.js

```diff
@@ -173,8 +173,9 @@ var LibraryManager = {
     for (var i = 0; i < libraries.length; i++) {
       var filename = libraries[i];
       var src = read(filename);
+      var processed = undefined;
       try {
-        var processed = processMacros(preprocess(src, filename));
+        processed = processMacros(preprocess(src, filename));
```

Comment:
hmm, this shouldn't matter, I don't think: the scope of `var processed` is the whole function, not just the `try` block, and it's set to `undefined` before anyhow?

Reply:
Sorry I don't quite understand what you mean here, but I've just updated my commit message to make it more clear. Please take another look.

Reply:
Sorry, I don't understand the commit message. There must be some underlying assumption one of us is making that is confusing the other.

Can you create a concrete testcase that shows what this fixes? If there is code I can run and see how it is fixed, that may help me see what you mean.

For my part, unless I am missing something here, there should be no change to behavior from this PR: `processed` has function scope before and after this PR, and it has an initial value of undefined before and after this PR as well, so I don't see what this changes: everything is still the same, isn't it?

Reply:
Please try the following case:

```bash
cat << EOF > main.c
int main()
{
  return 0;
}
EOF
cat << EOF > library_foo.js
mergeInto(LibraryManager.library, {
  foo: function () {
    var bar = {{{ invalid js here }}};
  },
});
EOF
emcc main.c --js-library library_foo.js
```

library_foo.js should be printed in the error message, but library_vr.js is printed instead.

Reply:
Thanks! Now I see. JS is confusing... but you're right, this is necessary. `processed` can hold the old value from the previous iteration, if we don't manually assign something to it.